src/handwritten/_install_daqmx.py

```diff
@@ -0,0 +1,332 @@
+        logging.debug(traceback.format_exc())
+        raise click.ClickException(f'Invalid version format found') from e
+
+def _is_latest_version_greater(latest_version : str, installed_version: str) -> int:
+    """
+    Compare two versions which are in the format: major.minor.update
+
+    Args:
+        installed_version (str): The installed version to compare.
+        latest_version (str): The version to compare against.
+
+    Returns:
+        int: False if versions are equal and installed_version is greater than latest_version
+        - True if installed_version is less than latest_version,
+             .
+
+    >>> _is_latest_version_greater("23.8.0", "23.8.0")
+    Latest Version already installed
+    False
+    >>> _is_latest_version_greater("23.8.1", "24.0.0")
+    Latest Version already installed
+    False
+    >>> _is_latest_version_greater("23.8.1", "23.8.0")
+    True
+    >>> _is_latest_version_greater("24.0.0", "23.8.5")
+    True
+    >>> _is_latest_version_greater("invalid_version", "1.2.0")
+    Traceback (most recent call last):
+    ...
+    click.exceptions.ClickException: Invalid version format found
+    """
+    try:
+        logging.debug("Entering _compare_version")
+        latest_parts: Tuple[int, ...] = _parse_version(latest_version)
+        installed_parts: Tuple[int, ...] = _parse_version(installed_version)
+
+        if latest_parts > installed_parts:
+            return True
+        else:
+            print(f"Latest Version already installed")
+            return False
+    except ValueError  as e:
+        logging.info("Error: %s", e)
+        logging.debug(traceback.format_exc())
+        raise click.ClickException(f'Invalid version format found') from e
+
+
+def _get_daqmx_installed_version() -> str:
+    """
+    Check for existing installation of DAQmx by checking for Version information.
+    Version information is read from the registry key.
+    
+    Returns:
+        str: The version of the installed NI-DAQmx if found, else an empty string.
+
+    >>> _get_daqmx_installed_version()  # doctest: +SKIP
+    ''
+    """
+    try:
+        logging.debug("Reading the registry entries to get installed DAQmx version")
+        with winreg.OpenKeyEx(
+            winreg.HKEY_LOCAL_MACHINE,
+            r"SOFTWARE\National Instruments\NI-DAQmx\CurrentVersion",
+            0,
+            winreg.KEY_READ | winreg.KEY_WOW64_32KEY,
+        ) as daqmx_reg_key:
+            product_name = winreg.QueryValueEx(daqmx_reg_key, "ProductName")[0]
+            product_version = winreg.QueryValueEx(daqmx_reg_key, "Version")[0]
+
+        if product_name == "NI-DAQmx":
+            logging.debug("Found registry entries for Product Name: %s and version %s", product_name , product_version )
+            return product_version
+    except FileNotFoundError:
+        logging.info("No existing NI-DAQmx installation found.")
+        logging.debug(traceback.format_exc())
+        return None
```

Comment:
`None` is not a `str`. Either return `""` or change the type hint to `-> Optional[str]`.